.github/workflows/build.yml

```diff
@@ -11,9 +11,27 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: cachix/install-nix-action@v25
+      - name: Setup Cachix
+        uses: cachix/cachix-action@v14
+        with:
+          name: devenv
+      - name: Install devenv
+        run: nix-env -if https://install.devenv.sh/latest
+      - name: Run checks
+        run: devenv ci
```

Comment:
It's not immediately obvious to be what this job is doing by the name, but I'm not super familiar with `devenv ci` so I'll defer to you on this one. I guess it would be clear if there was a formatting error, but wanted to bring it up in case you can think of a better name.